scripts/system_monitoring_functions/diane.sh

```
@@ -0,0 +1,241 @@
+#!/bin/bash
+
+# =======================================
+# HIDS Toolkit - System Resource Monitor
+# Monitors CPU, Memory, Disk, and Swap.
+# =======================================
+
+# Source the configuration file for thresholds and log paths
+source "./toolkit/config.sh"
```

Shellcheck (0.10.0)

[error] Literal carriage return. Run script through tr -d '\r' .

(SC1017)